src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ProviderSettingsViewModel.cs

```diff
@@ -90,4 +147,21 @@ private List<TopLevelViewModel> BuildTopLevelViewModels()
     }
 
     private void Save() => SettingsModel.SaveSettings(_settings);
+
+    private void InitializeSettingsPage()
+    {
+        if (_provider.Settings == null)
+        {
+            return;
+        }
+
+        _provider.Settings.SafeInitializeProperties();
+        _provider.Settings.DoOnUiThread(() =>
```

Comment:
What does doing this on the UI thread get us? (potentially good opportunity for a comment)

Reply:
Betting due to invoking the property changed event

Reply:
Clint is right here